### PR TITLE
feat: thorchain lp deposit jit halted checks

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR:

- adds an "Available" tag to available (not halted) pools
- add pool halted JIT checks at
  - Add Liquidty input
  - Transaction confirm both when rendering, and after clicking signing, right before the Tx is signed and broadcasted (which it now won't if halted) 
  - Pool page - disabled routing to "Add Liquidity" page with a tooltip if the pool is currently halted
  
  
## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to https://github.com/shapeshift/web/issues/6108

> What protocols, transaction types or contract interactions might be affected by this PR?

Low, though pools could be wrongly detected as halted, ensure that's not the case
 
## Testing

- Ensure active pools are shown as active, and halted as halted in "Available Pools" page (DOGE is currently halted at the time of writing)
- Ensure it is not possible to click "Add Liquidity" from an halted pool page
- Ensure "Pool Halted" is shown as a submit button status when the pool is halted

### Engineering

⚠️  note: while this does bring the JIT logic and view-layer for LP halted checks, freshness is not guaranteed in this PR.
See the one on top of it in this stack (https://github.com/shapeshift/web/pull/6164) with the fix.

- ☝🏽 
- The signing flow can't be tested as it's obviously very edge-casy, but you can monkey patch it two ways:

1. monkey patch `isTradingActive` to false, which will simulate the trade going to halted while rendering the Tx signing step for a given Tx
2. monkey patch the query itself to return false so that `<TransactionRow />` shows the pool halted state, e.g with the diff below. Then click sign and notice the pool goes halted

```diff
diff --git a/src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx b/src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx
index e3ddf2cfea..436044a758 100644
--- a/src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx
+++ b/src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx
@@ -79,6 +79,9 @@ type TransactionRowProps = {
   asymSide?: AsymSide | null
 }
 
+let count = 0
+const mockIsTradingActive = () => !count
+
 export const TransactionRow: React.FC<TransactionRowProps> = ({
   assetId,
   poolAssetId,
@@ -105,12 +108,13 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
     isLoading: isTradingActiveLoading,
     isRefetching: isTradingActiveRefetching,
     refetch: refetchIsTradingActive,
-  } = useQuery(
-    reactQueries.common.isTradingActive({
+  } = useQuery({
+    ...reactQueries.common.isTradingActive({
       assetId: poolAssetId,
       swapperName: SwapperName.Thorchain,
     }),
-  )
+    select: data => mockIsTradingActive(),
+  })
 
   const runeAccountId = accountIdsByChainId[thorchainChainId]
   const poolAssetAccountId =
@@ -390,6 +394,7 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
 
     return (async () => {
       // Refetch the trading active state JIT to ensure the pool didn't just become halted
+      count++
       const { data: isTradingActiveData } = await refetchIsTradingActive()
       if (!isTradingActiveData) throw new Error('Pool Halted')
 

```

### Operations

- ☝🏽 
## Screenshots (if applicable)

image
image
image
image

- Monkey-patched pool went halted at signing time

https://github.com/shapeshift/web/assets/17035424/917ba40c-b6d5-47e2-a15e-e4719699198a

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6146?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #6138** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6138?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6139** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6139?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #6146** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6146?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6146?utm_source=stack-comment).

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -955,6 +969,7 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
         <TradeAssetSelect
           assetId={thorchainAssetId}
           onAssetChange={handleAssetChange}
+          isReadOnly
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:972]: Not strictly related but this was spotted by @MBMaria - sneaking this in this PR

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 2]
```diff
@@ -614,7 +631,13 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
             size='lg'
             colorScheme='blue'
             onClick={handleSignTx}
-            isLoading={status === TxStatus.Pending || isInboundAddressLoading}
+            isDisabled={isTradingActive === false}
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:634]: This needs a copy for halted

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:641]: https://github.com/shapeshift/web/pull/6146/files#diff-29561149fe9fe8c5a89defe8be8fcfb91492b083ce61cf186b981069f24724c6R588

**Review by 0xApotheosis — APPROVED** [review]
Confirmed that active LPs are still happy and can be interacted with, and monkey patched to confirm that add liquidity confirm and the add liquidity page are correctly disabled.
Screenshot 2024-02-12 at 8 36 34 pm

Screenshot 2024-02-12 at 8 36 23 pm


Note to self, we'll want to add this to the "Remove Liquidity" flow, too.

**Comment by 0xApotheosis** [comment]
and gm for the brilliant testing information, @gomesalexandre!

**Pull request merged**